src/tcp.c

```diff
@@ -32,6 +32,22 @@
 
 #define get_sin_port(addr) (((struct sockaddr_in*) addr->ai_addr)->sin_port)
 
+void
+enftun_tcp_native_init(struct enftun_tcp_native* ctx,
+                       struct enftun_tcp* sock,
+                       int mark)
+{
+    ctx->socket = *sock;
+
+    ctx->socket.ops.connect     = (int (*)(void*, const char* host,
```

Comment:
Instead of assigning each function pointer in `_init`, create a static instance
```
static struct enftun_tcp_ops enftun_tcp_ops_native = {
   . connect = enftun_tcp_native_connect,
   . connect_any = ...,
   .close = ...
};
```

And then just assign that in the `_init` function
```
ctx->base.ops = enftun_tcp_ops_native
```